google/cloud/bigquery/client.py

```diff
@@ -253,16 +257,18 @@ def __init__(
             if client_options.api_endpoint:
                 api_endpoint = client_options.api_endpoint
                 kw_args["api_endpoint"] = api_endpoint
-            elif (
-                hasattr(client_options, "universe_domain")
-                and client_options.universe_domain
-                and client_options.universe_domain is not _DEFAULT_UNIVERSE
-            ):
-                kw_args["api_endpoint"] = _DEFAULT_HOST.replace(
-                    _DEFAULT_UNIVERSE, client_options.universe_domain
-                )
+            else:
+                client_universe = _get_client_universe(client_options)
+                if client_universe != _DEFAULT_UNIVERSE:
+                    kw_args["api_endpoint"] = _DEFAULT_HOST_TEMPLATE.replace(
+                        "{UNIVERSE_DOMAIN}", client_universe
+                    )
+        # Ensure credentials and universe are not in conflict.
+        if hasattr(self, "_credentials") and client_universe is not None:
+            _validate_universe(client_universe, self._credentials)
 
         self._connection = Connection(self, **kw_args)
+
```

Comment:
nit: should we keep the blank line here?

Reply:
removed